lib/qxcli/commands/Command.js

```diff
@@ -128,37 +129,43 @@ qx.Class.define("qxcli.commands.Command", {
           throw new Error(`Invalid manifest file ${manifestPath}.`);
         }
       };
+      if ( ! qxpath ) {
+        throw new Error( "Cannot determine qooxdoo path." );  
+      }
       return path.join(appPath,qxpath);
     },
 
 
     /**
      * Returns the absolute path to the qooxdoo framework used by the current project, unless
-     * the user provided an option "qxpath" in the argv Map, in which case this value is returned.
-     * @return {Promise(String)|false} Promise that resolves with the absolute path {String} or false 
-     * if no path can be determined. 
+     * the user provided a CLI option "qxpath", in which case this value is returned.
+     * @throws {Error} if no path can be determined
+     * @return {Promise(String)} Promise that resolves with the absolute path 
      */    
-    getUserQxPath : async function(argv={}){
-      let qxpath = (argv.qxpath !== undefined) ? argv.qxpath : await this.getAppQxPath();
+    getUserQxPath : async function(){
+      let qxpath = (this.argv.qxpath !== undefined) ? this.argv.qxpath : await this.getAppQxPath();
       return path.isAbsolute(qxpath) ? qxpath : path.resolve( qxpath );
     },
 
     /**
-     * Returns the path to the qooxdoo framework used by the current project. To be replaced by
-     * getUserQxPath()
-     * @return {Promise} Promise that resolves with the path {String}
-     */
-    getQooxdooPath : async function(argv)
-    {
-      return await this.getUserQxPath(argv);
-      //const cfg_path = path.join(process.cwd(), 'compile.json');
-      //let maker = await new qxcli.commands.Compile(argv).configure(cfg_path);
-      //return maker.getAnalyser().getQooxdooPath();
+     * Returns the version of the qooxdoo framework used by the curret project, or of the qooxdoo framework
+     * given by the "qxpath" CLI option.
+     * @throws {Error} If the version cannot be determined
+     * @return {Promise(String)} Promise that resolves with the version string
+     */ 
+    getUserQxVersion : async function(){ 
+      let qxpath = await this.getUserQxPath();
+      let qxversion = await fs.readFileAsync( path.join( path.dirname(qxpath), 'version.txt' ), 'utf-8' );
+      if( ! semver.valid( qxversion) ){
+        throw new Error(`Invalid qooxdoo version ${qxversion}. Must be valid semver string.`);
+      }
+      return qxversion;
     },
     
     /**
      * Given the path to a qooxdoo framework folder, returns the qooxdoo version 
      * in semver-compatible format.
+     * @deprecated Use {@link #getUserQxVersion} instead.
```

Comment:
I agree with @hkollmann there's no need to deprecate at this stage, we're still officially in beta and the API is allowed to change without warning.